rmw_connext_xtypes_dynamic_cpp/src/functions.cpp

```diff
@@ -907,18 +1172,98 @@ rmw_create_service(
+    rmw_set_error_string("node handle not from this implementation");
+    // printf("but from: %s\n", node->implementation_identifier);
+    return NULL;
+  }
+
+  DDSDomainParticipant * participant = static_cast<DDSDomainParticipant *>(node->data);
+
+  if (type_support->typesupport_identifier !=
+    rosidl_typesupport_introspection_cpp::typesupport_introspection_identifier)
+  {
+    rmw_set_error_string("type support not from this implementation");
+    // printf("but from: %s\n", type_support->typesupport_identifier);
+    return NULL;
+  }
+
+  const rosidl_typesupport_introspection_cpp::ServiceMembers * service_members =
+    (rosidl_typesupport_introspection_cpp::ServiceMembers *)type_support->data;
+  const rosidl_typesupport_introspection_cpp::MessageMembers * request_members =
+    service_members->request_members_;
+  const rosidl_typesupport_introspection_cpp::MessageMembers * response_members =
+    service_members->response_members_;
+  std::string request_type_name = std::string(request_members->package_name_) + "::dds_::" +
+    request_members->message_name_ + "_";
+  std::string response_type_name = std::string(response_members->package_name_) + "::dds_::" +
+    response_members->message_name_ + "_";
+
+  DDS_DomainParticipantQos participant_qos;
+  DDS_ReturnCode_t status = participant->get_qos(participant_qos);
+  if (status != DDS_RETCODE_OK) {
+    rmw_set_error_string("failed to get participant qos");
+    // printf("get_qos() failed. Status = %d\n", status);
+    return NULL;
+  }
+
+  DDS_TypeCode * request_type_code = create_type_code(
+    request_type_name, request_members, participant_qos);
+  DDS::DynamicDataTypeSupport * request_type_support = new DDS::DynamicDataTypeSupport(
+    request_type_code, DDS_DYNAMIC_DATA_TYPE_PROPERTY_DEFAULT);
+
+  DDS_TypeCode * response_type_code = create_type_code(
+    response_type_name, response_members, participant_qos);
+  DDS::DynamicDataTypeSupport * response_type_support = new DDS::DynamicDataTypeSupport(
+    response_type_code, DDS_DYNAMIC_DATA_TYPE_PROPERTY_DEFAULT);
+
+  // create requester
+  connext::ReplierParams<DDS_DynamicData, DDS_DynamicData> replier_params(participant);
+  replier_params.service_name(service_name);
+  replier_params.request_type_support(request_type_support);
+  replier_params.reply_type_support(response_type_support);
+
+  connext::Replier<DDS_DynamicData, DDS_DynamicData> * replier(
+    new connext::Replier<DDS_DynamicData, DDS_DynamicData>(replier_params));
+
+  DDSDataReader * request_datareader = replier->get_request_datareader();
+
+  ConnextDynamicServiceInfo * server_info = new ConnextDynamicServiceInfo();
+  server_info->replier_ = replier;
+  server_info->request_datareader_ = request_datareader;
+  server_info->response_type_support_ = response_type_support;
+  server_info->request_members_ = request_members;
+  server_info->response_members_ = response_members;
+
+  rmw_service_t * service = new rmw_service_t;
```

Comment:
Added in f8f5e8be2bf0d004a10ad729b1a4235f056465b0 - also for the static Connext implementation.